python/dgllife/model/model_zoo/potentialnet.py

```diff
@@ -0,0 +1,223 @@
+import math
+import dgl
+import numpy as np
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl import function as fn
+from torch.nn import init
+
+__all__ = ['PotentialNet']
+
+def sum_ligand_features(h, batch_num_nodes):
+    """
+    Computes the sum of only ligand features `h` according to the batch information `batch_num_nodes`.
+    """
+    node_nums = th.cumsum(batch_num_nodes, dim=0)
+    B = int(len(batch_num_nodes)/2) # actual batch size
+    ligand_idx = [list(range(node_nums[0]))] # first ligand
+    for i in range(2,len(node_nums),2): # the rest of ligands in the batch
+        ligand_idx.append(list(range(node_nums[i-1],node_nums[i])))
+    return th.cat([h[i,].sum(0, keepdim=True) for i in ligand_idx]).to(device=h.device) # sum over each ligand
+
+
+class PotentialNet(nn.Module):
+    """
+    Protein-ligand binding affnity prediction using a 'staged gated graph neural network'
+    introduced in `PotentialNet for Molecular Property Prediction <http://dx.doi.org/10.1021/acscentsci.8b00507>`__.
+
+    Parameters
+    ----------
+    f_in: int
+        The dimension size of input features to GatedGraphConv, 
+        equivalent to the dimension size of atomic features in the molecular graph.
+    f_bond: int
+        The dimension size of the output from GatedGraphConv in stage 1,
+        equivalent to the dimension size of input to the linear layer at the end of stage 1.
+    f_spatial: int
+        The dimension size of the output from GatedGraphConv in stage 2,
+        equivalent to the dimension size of input to the linear layer at the end of stage 2.
+    f_gather: int
+        The dimension size of the output from stage 1 & 2,
+        equivalent to the dimension size of output from the linear layer at the end of stage 1 & 2.
+    n_etypes: int
+        The number of heterogeneous edge types for stage 2.
+        Currently implemented as 5(the number of covalent bond types in stage 1) + the number of distance bins in stage 2.
```

Comment:
Why is this 5? Single, double, triple, aromatic. What's the one missing?

Reply:
https://lifesci.dgl.ai/generated/dgllife.utils.CanonicalBondFeaturizer.html#dgllife.utils.CanonicalBondFeaturizer
featurizes 12 originally. Then 7 of those edge types do not appear in the dataset

Reply:
But these are not edge types right? E.g. an edge can be single and conjugated simultaneously?